src/memory.rs

```diff
@@ -82,12 +82,15 @@ impl U8SliceView {
     }
 }
 
-/// A Vector type that requires explicit creation and destruction and
-/// can be sent via FFI.
+/// An optional Vector type that requires explicit creation and destruction
```

Comment:
Rrrr... the comments above belong here

Reply:
> Do we ever want Go to consume this?
> Do you expose methods to read it on the Go side?
>
> You can suggest using a different type (U8Slice) to pass into from Rust -> Go

Yes, sometimes binary data is passed from Rust to Go as the last thing of a call, e.g. the return type of `load_wasm`. When Go gets this, it must do the cleanup. No Rust code is running after that return which could destroy things. In those cases we call `copyAndDestroyUnmanagedVector` in Go, which does a copy any calls `destroy_unmanaged_vector`.

Reply:
One could argue what "consume" even means and if this higher level concept is helping or confusing.

Reply:
Thank you for that use cases. I was looking at those "return" arguments for eg. Next.

Of course the return value of handle must be this type too.

Explaining where and how it is used really helps me understand the structure better